hpke_test.go

```diff
@@ -450,6 +474,32 @@ func (rtt roundTripTest) Test(t *testing.T) {
 	exportedI := ctxI.Export(exportContext, exportLength)
 	exportedR := ctxR.Export(exportContext, exportLength)
 	assertBytesEqual(t, suite, "Incorrect exported secret", exportedI, exportedR)
+
+	// Verify encryption context serialization functionality
+	opaqueI, err := ctxI.Marshal()
+	if err != nil {
+		t.Fatalf("[%04x, %04x, %04x] Error serializing encrypt context: %v", rtt.kem_id, rtt.kdf_id, rtt.aead_id, err)
+	}
+
+	unmarshaledI, err := UnmarshalEncryptContext(opaqueI)
+	if err != nil {
+		t.Fatalf("[%04x, %04x, %04x] Error serializing encrypt context: %v", rtt.kem_id, rtt.kdf_id, rtt.aead_id, err)
+	}
+
+	assertCipherContextEqual(t, suite, "Encrypt context serialization mismatch", ctxI.cipherContext, unmarshaledI.cipherContext)
+
+	// Verify decryption context serialization functionality
+	opaqueR, err := ctxR.Marshal()
+	if err != nil {
+		t.Fatalf("[%04x, %04x, %04x] Error serializing decrypt context: %v", rtt.kem_id, rtt.kdf_id, rtt.aead_id, err)
+	}
+
+	unmarshaledR, err := UnmarshalDecryptContext(opaqueR)
+	if err != nil {
+		t.Fatalf("[%04x, %04x, %04x] Error serializing decrypt context: %v", rtt.kem_id, rtt.kdf_id, rtt.aead_id, err)
+	}
+
+	assertCipherContextEqual(t, suite, "Decrypt context serialization mismatch", ctxR.cipherContext, unmarshaledR.cipherContext)
```

Comment:
Can we test that `bytes.Equal(unmarshaledR.Export(exportContext, exportLength), exportedR)`?